OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -1399,8 +1575,12 @@ public static DreamValue NativeProc_shutdown(DreamObject instance, DreamObject u
         [DreamProcParameter("Text", Type = DreamValueType.String)]
         [DreamProcParameter("File", Type = DreamValueType.String)]
         public static DreamValue NativeProc_text2file(DreamObject instance, DreamObject usr, DreamProcArguments arguments) {
-            string text = arguments.GetArgument(0, "Text").GetValueAsString();
-            string file = arguments.GetArgument(1, "File").GetValueAsString();
+            if(!arguments.GetArgument(0, "Text").TryGetValueAsString(out var text) ||
+               !arguments.GetArgument(1, "File").TryGetValueAsString(out var file))
+            {
+                //TODO Check if this is the correct behavior
+                return DreamValue.Null;
```

Comment:
If text is not a string then empty text is written to the file. If the file is null, `0` is returned.